File: lab2-functionalprimer/python3/func_as_arg.py
def listFunc(start, end):
    return [i for i in range(start, end+1)] #Create list of ints from 1 to 5, Haskell equivalent [1..5]

def applicatorFunc(inpFunc, s):
    if s=='s':
        return sum(inpFunc)
    else:
        return sum(inpFunc)/len(inpFunc)

print(applicatorFunc(listFunc(1, 5), 's'))

File: lab2-functionalprimer/javascript/func_as_arg.js
//Create list of ints from 1 to 5, Haskell equivalent [1..5]
function arrFunc(start, end){
    let arr = [];
    for (let i = start; i<=end; i++) {
        arr.push(i);
    }
    return arr;    
}

function applicatorFunc(inpFunc, s){
    if(s=='s'){
        const arr = inpFunc;        
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum;
    }
    else{        
        const arr = inpFunc;
        let sum = arr.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
        return sum/inpFunc.length;
    }
}

let x = applicatorFunc(arrFunc(1, 5), 's');
console.log(x);